File: TDD/PhaseGate/PhaseGateDay6/RandomSubtractionProblems.java
import java.util.Scanner;
public class RandomSubtractionProblems{
    public static void main(String[] args){
    Scanner input = new Scanner(System.in);
        int programRun = 0;
        int numOfTrials = 0;
        int countCorrect = 0;
        int countWrong = 0;
        int wrong = 0;
        double ans = 0;
        while(programRun != 10){
            double num1 = Math.floor(Math.random() * 100);
            double num2 = Math.floor(Math.random() * 100);
            if(programRun > 9){
                break;
            }
            else{
                if(num1 < num2){
                    double temp = num1;
                    num1 = num2;
                    num2 = temp;
                }
                double result = num1 - num2;
          //  for(let trials = 0; trials <= 9; trials++){
           //     const prompt = require("prompt-sync")();
                
                System.out.print("What is " + num1 + " - " + num2 + "?: ");
                ans = input.nextInt();
                    if(ans == result){
                        System.out.print(ans + " is correct.");
                        countCorrect = countCorrect + 1;
                    }
                    else if(ans != result){
                        for(wrong = 0; wrong < 1; wrong++){
                            System.out.print("Wrong. One trial left!");
                            System.out.print("What is " + num1 + " - " + num2 + "?:");
                            ans = input.nextInt();
                        }
                        countWrong = countWrong + 1;
                    }
                System.out.print(countCorrect + " correct score!");
                System.out.print(countWrong + " wrong score!");
                programRun = programRun + 1;
            }
        }

    }
}

File: JavaScript/phaseGateDay6/RandomSubtractionProblems.js
let num1 = Math.floor(Math.random() * 100);
let num2 = Math.floor(Math.random() * 100);
let numOfTrials = 0;
let countCorrect = 0;
let countWrong = 0;
let wrong = 0;
let ans = 0;
    if(num1 < num2){
        let temp = num1;
        num1 = num2;
        num2 = temp;
    }
    let result = num1 - num2;
for(let trials = 0; trials <= 9; trials++){
    const prompt = require("prompt-sync")();
    let ans = prompt("What is " + num1 + " - " + num2 + "?: ");
        if(ans == result){
            console.log(ans + " is correct.");
            countCorrect = countCorrect + 1;
        }
        else if(ans != result){
            for(wrong = 0; wrong < 1; wrong++){
                console.log("Wrong. One trial left!");
                const prompt = require("prompt-sync")();
                let ans = prompt("What is " + num1 + " - " + num2 + "?:");
            }
            countWrong = countWrong + 1;
        }
    console.log(countCorrect + " correct score!");
    console.log(countWrong + " wrong score!");
}
